x/ibc/core/23-commitment/types/merkle.go

```diff
@@ -68,51 +68,53 @@ func (mp MerklePrefix) Empty() bool {
 var _ exported.Path = (*MerklePath)(nil)
 
 // NewMerklePath creates a new MerklePath instance
-func NewMerklePath(keyPathStr []string) MerklePath {
-	merkleKeyPath := KeyPath{}
-	for _, keyStr := range keyPathStr {
-		merkleKeyPath = merkleKeyPath.AppendKey([]byte(keyStr), URL)
-	}
-
+// The keys must be passed in from root-to-leaf order
+func NewMerklePath(keyPath ...string) MerklePath {
+	kp := keyPath[:]
 	return MerklePath{
-		KeyPath: merkleKeyPath,
+		KeyPath: kp,
 	}
 }
 
 // String implements fmt.Stringer.
 func (mp MerklePath) String() string {
-	return mp.KeyPath.String()
+	return "/" + strings.Join(mp.KeyPath, "/")
 }
 
 // Pretty returns the unescaped path of the URL string.
 func (mp MerklePath) Pretty() string {
-	path, err := url.PathUnescape(mp.KeyPath.String())
+	path, err := url.PathUnescape(mp.String())
 	if err != nil {
 		panic(err)
 	}
 	return path
 }
 
+// GetKey will return a byte representation of the key
+// after URL escaping the key element
+func (mp MerklePath) GetKey(i uint64) ([]byte, error) {
+	key, err := url.PathUnescape(mp.KeyPath[i])
+	if err != nil {
+		return nil, err
+	}
+	return []byte(key), nil
+}
+
 // Empty returns true if the path is empty
 func (mp MerklePath) Empty() bool {
-	return len(mp.KeyPath.Keys) == 0
+	return len(mp.KeyPath) == 0
 }
 
 // ApplyPrefix constructs a new commitment path from the arguments. It interprets
 // the path argument in the context of the prefix argument.
 //
 // CONTRACT: provided path string MUST be a well formated path. See ICS24 for
 // reference.
-func ApplyPrefix(prefix exported.Prefix, path string) (MerklePath, error) {
```

Comment:
This did not match the spec here: https://github.com/cosmos/ics/blob/master/spec/ics-023-vector-commitments/README.md

So I changed interface to match spec